src/protocols/output_management.rs

```diff
@@ -0,0 +1,774 @@
+                    }
+                    client_data.confs.insert(conf);
+                } else {
+                    error!("CreateConfiguration: missing client data");
+                }
+            }
+            zwlr_output_manager_v1::Request::Stop => {
+                let clients = &mut state.output_management_state().clients;
+                clients.get(&client.id()).map(|c| c.manager.finished());
+                clients.remove(&client.id());
+            }
+            _ => unreachable!(),
+        }
+    }
+}
+
+impl<D> Dispatch<ZwlrOutputConfigurationV1, OutputConfigurationData, D>
+    for OutputManagementManagerState
+where
+    D: Dispatch<ZwlrOutputConfigurationV1, OutputConfigurationData>,
+    D: Dispatch<ZwlrOutputHeadV1, String>,
+    D: Dispatch<ZwlrOutputConfigurationHeadV1, OutputConfigurationHeadState>,
+    D: OutputManagementHandler,
+    D: 'static,
+{
+    fn request(
+        state: &mut D,
+        client: &Client,
+        conf: &ZwlrOutputConfigurationV1,
+        request: zwlr_output_configuration_v1::Request,
+        conf_data: &OutputConfigurationData,
+        _display: &DisplayHandle,
+        data_init: &mut DataInit<'_, D>,
+    ) {
+        let g_state = state.output_management_state();
+        let outdated = conf_data.lock().unwrap().serial != g_state.serial;
+        match request {
+            zwlr_output_configuration_v1::Request::EnableHead { id, head } => {
+                let head_name = match head.data::<String>() {
+                    Some(name) => name,
+                    None => {
+                        error!("EnableHead: Missing attached output head name");
+                        let _fail = data_init.init(id, OutputConfigurationHeadState::Cancelled);
+                        return;
+                    }
+                };
+                if outdated {
+                    info!("EnableHead: request from an outdated configuration");
+                    let _fail = data_init.init(id, OutputConfigurationHeadState::Cancelled);
+                    return;
+                }
+                let mut conf_data = conf_data.lock().unwrap();
+                let output = match conf_data.new_out.get_mut(head_name) {
+                    Some(out) => out,
+                    None => {
+                        error!("EnableHead: output configuration missing requested new head",);
+                        let _fail = data_init.init(id, OutputConfigurationHeadState::Cancelled);
+                        return;
+                    }
+                };
+                if output.current_mode.is_none() {
+                    output.current_mode = output.modes.iter().position(|a| a.is_preferred);
+                }
+                let current_mode = match output.current_mode {
+                    Some(current) => current,
+                    None => {
+                        error!("EnableHead: missing output prefered mode");
+                        return;
```

Comment:
A Wayland new ID must always be initialized with `data_init.init()`. This code path is missing it, probably where the errors are coming from.